### PR TITLE
5465 rows per page doesnt update on import assets

### LINKED ISSUE #5465: iSC/Cold chain/GAPS: Rows per page does not get updated/save when importing cold chain equipment
## What went wrong? 😲

Rows per page selection does not get updated/save when importing cold chain equipment!

https://github.com/user-attachments/assets/31cf3cd8-e9a9-4c76-8dc8-d7dd57d8482e

At first, I thought maybe it doesn't work because there are less than 20 entries, but here is an expected behaviour where I have 8 entries but still able to change how many are displayed on the page

https://github.com/user-attachments/assets/6f1b8eab-e5d7-4fdc-8891-c112bc02a30a

## Expected behaviour 🤔

Should be able to save the selection

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to cold chain -> Equipment
2. Import -> select a file (if you don't have a sample, you can export the assets you have and then import the same file for ease of testing)
3. Should see a modal with items imported -> try changing how many you see per page
4. See error!

## Your environment 🌱

- Open mSupply Version: V2.4.0-RC1
- Legacy mSupply Central Server Version: V7.18.11

- Platform: Google Chrome (browser for Mac)

- Database type: SQLite3

### PR DESCRIPTION
**Pull request by aimee-mcneil-melville**
Fixes #5465 

# 👩🏻‍💻 What does this PR do?

Allows for correct pagination  in the rows per page displayed on the import asset modal - both in the equipment section and catalogue assets section. 

This uses the user default settings, and allows the user to also select a number of rows to display from the dropdown.

I made a new hook to accomadate this, as it is a modal on top of a List View that also has pagination (stored in url)

This allows a user to use the pagination function when importing assets, by first using their default settings and the user can change within the modal as needed

## 💌 Any notes for the reviewer?

A couple of things I noticed but not within scope of this issue, I can raise issues for these separately

Styling:
- there is a secondary scroll bar within the table, as well as at the edge of the modal. this impacts scrolling ease
- Table header for Catalogue Item Code is on three vertical lines

Screenshot 2024-11-28 at 5 12 21 PM

There is another issue #3587 to address the duplication of these two components

# 🧪 Testing

- [ ] Go to cold chain -> Equipment
- [ ] Import -> add a file to import (I have a sample of 103 assets which I can share)
- [ ] Should see a modal with items imported -> try changing how many you see per page
- [ ] The rows should update to display the selected number of rows per page, and change to reflect which page is selected both when choosing a page or using next/previous arrows

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  4.

### CODE REVIEW

File: client/packages/coldchain/src/Equipment/ImportAsset/ImportReviewDataTable.tsx [hunk 1]
```diff
@@ -24,11 +24,14 @@ export const ImportReviewDataTable: FC<ImportReviewDataTableProps> = ({
...
-    first: 20,
-    offset: 0,
-  });
+
+  const {
+    pagination: { page, first, offset },
+    updateUserPreferencePagination,
+  } = useUserPreferencePagination();
+
+  const pagination = { page, first, offset };
```
Comment (lache-melvin) [inline client/packages/coldchain/src/Equipment/ImportAsset/ImportReviewDataTable.tsx:33]:
```suggestion
  const {
    pagination,
    updateUserPreferencePagination,
  } = useUserPreferencePagination();
```

Looks like you're pulling these fields off just to rebuild an identical object - should be all good to pass pagination through directly!

File: client/packages/system/src/Asset/ImportCatalogueItem/ImportReviewDataTable.tsx [hunk 2]
```diff
@@ -18,11 +18,14 @@ export const ImportReviewDataTable: FC<ImportReviewDataTableProps> = ({
...
-    first: 20,
-    offset: 0,
-  });
+
+  const {
+    pagination: { page, first, offset },
+    updateUserPreferencePagination,
+  } = useUserPreferencePagination();
+
+  const pagination = { page, first, offset };
```
Comment (lache-melvin) [inline client/packages/system/src/Asset/ImportCatalogueItem/ImportReviewDataTable.tsx:27]: and same here :)

**Review by lache-melvin — COMMENTED** [review]
Looking good Aimee! Just would be good to remove that extra destructure/recreate, and then to do something about the table height quickly...

> there is a secondary scroll bar within the table, as well as at the edge of the modal. this impacts scrolling ease

[Here is an example of a quick fix](https://github.com/msupply-foundation/open-msupply/compare/5465-rows-per-page-doesnt-update-on-import-assets...5465-rows-per-page-doesnt-update-on-import-assets(table-height)) - tables need a fixed height or they just keep expanding ... not perfect but might be better than nothing for this PR? If you're happy with it, feel free to apply to other table as well?


> Table header for Catalogue Item Code is on three vertical lines

Ah yeah, this isn't great but will be part of bigger table refactor in future 🙏

**Comment by aimee-mcneil-melville** [comment]
Great, changes made!
Scrolling is definitely smoother and more user friendly

**Review by lache-melvin — APPROVED** [review]
Lovely thank you! 🚀

**Pull request merged**